includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -755,6 +761,41 @@ private function has_used_attributes( $attribute_names ) {
 		return true;
 	}
 
+	/**
+	 * Whether a given class is allowed to be styled in <amp-date-picker>.
+	 *
+	 * That component has child classes that won't be present in the document yet.
+	 * So get whether a class is an allowed child.
+	 *
+	 * @since 1.5.0
+	 * @link https://github.com/airbnb/react-dates/tree/05356/src/components
+	 *
+	 * @param string $class The name of the class to evaluate.
+	 * @return bool Whether the class is allowed as a child of <amp-date-picker>.
+	 */
+	private function is_class_allowed_in_amp_date_picker( $class ) {
+		// Some prefixes, some full class names.
+		$allowed_class_beginnings = [
+			'CalendarDay',
```

Comment:
This might be missing some classes, though I tried to be thorough.

I looked at the [components](https://github.com/airbnb/react-dates/tree/05356400baa8e05a9d7ea17e74fbb04bb3b23fa8/src/components) that have styles. 

This doesn't include components that are just an `.svg`, like [CalendarIcon](https://github.com/airbnb/react-dates/blob/05356400baa8e05a9d7ea17e74fbb04bb3b23fa8/src/components/CalendarIcon.jsx).

Reply:
Do any of these component names get carried over to be class names in the DOM? If so, they should be included in this list.

Reply:
Yes, all of these appear as names in the DOM except for `DateInput`, `DateRangePicker`, `DateRangePicker`:

<img width="669" alt="amp-date-picker" src="https://user-images.githubusercontent.com/4063887/75833804-c83ea580-5d7f-11ea-80bd-70915b54e143.png">

For those 3 that didn't appear, I couldn't find them, when using most of the snippets in the [amp-date-picker](https://amp.dev/documentation/examples/components/amp-date-picker/) documentation. I should probably remove those, though maybe I just missed a case where they are used.

Reply:
The 3 that I couldn't find in the DOM as class names are removed in https://github.com/ampproject/amp-wp/pull/4339/commits/5bffd65ecba76846919095ea883c93d6df20954a